### PR TITLE
Add CICD / Observability / Onboarding docs

### PR DESCRIPTION
**Pull request by ocervell**
Add documentation for OFF CICD process, Observability and Onboarding.

### CODE REVIEW

File: docs/cicd.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+The steps for executing a manual rollback are as follow:
+
+* SSH to the QEMU VM (either pre-production or production) and go to the deployment folder (usually named after the GitHub environment we are deploying to)
+* Replace the `TAG` variable by `sha-<COMMIT_SHA>` (where `COMMIT_SHA` is the last 'good' commit) in the `.env` file and restore it in the checked out repository.
+* Copy the `.env` file outside of the checked out repository so that it can be restored later.
+* Run `git checkout -qf <COMMIT_SHA>` of the last 'good' commit.
+
+Note that since deployments are automated, the following alternative also exists and is safer, although it can be a bit longer considering the git process:
+
+* Revert the 'bad' commit (`git revert <COMMIT_SHA>`) and make a pull request
```
Comment (alexgarel) [inline docs/cicd.md:138]: Another option, is to go restore to an old commit.

Like :
```
* restore locally to an old commit / tag you know is safe `git checkout <COMMIT_SHA_or_TAG>`
* create a new branch from there `git checkout -b deploy-<something>`
```

File: docs/cicd.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+* Push to a branch called `deploy-<something>` to deploy to pre-prod
+* **Merge the pull request**: the release workflow runs and creates a new release branch.
+* **Merge the release branch**: the revert will be deployed in production.
+
```
Comment (alexgarel) [inline docs/cicd.md:142]: I think we should add a note to remember that if there was some data/file migration, you have to figure out how to revert changes, if needed.

File: docs/cicd.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+## CICD status
+
+The current status of the automation of the deployment and testing processes across Open Food Facts repositories is as follows:
+
+| Repository                                                                            | Continuous Testing   | Continuous Deployment | Pre-production deployment | Production deployment                          | Release automation |
```
Comment (alexgarel) [inline docs/cicd.md:148]: Completely optional: you may add [emojis](https://gist.github.com/rxaviers/7360908), to help visualization, like `:heavy_check_mark: Automated`

(but maybe table will grow too wide…)

File: docs/cicd.md [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+Contact an OFF administrator to analyze why it is failing: the admin might have 
+to revert your PR to restore the previous working version in pre-production; you 
+can then continue to work on your branch to fix the problem, and make another PR.
+
+Ask the OFF administrator to deploy your PR before merging it, so that it is 
+known ahead of time if the PR will break the pre-production environment.
+
+### I don't have much confidence in the next release, can I make a release candidate before publishing the official release ?
+
+Yes, assuming your next version is `v1.1.0`, just create a git tag following semantic versioning using `git tag v1.1.0rc1 && git push --tags` and the automated process will deploy this release candidate to production.
```
Comment (alexgarel) [inline docs/cicd.md:191]: will deploy to production ? What is the point in making a rc then ?

File: docs/observability.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# Observability
+
+This document describes the observability stack used at Open Food Facts to monitor applications. 
+
+Having a good observability stack is critical to spend less time when debugging failures, to have a comprehension of how applications behave over time, and to have the ability to compare a software version with the previously deployed one.
+
+The observability stack used in the OFF stack is comprised of the following applications:
+
+* **Filebeat** as a logs collection agent deployed on each QEMU VM with Docker containers.
```
Comment (alexgarel) [inline docs/observability.md:9]:
```suggestion
* [**Filebeat**](https://www.elastic.co/guide/en/beats/filebeat/current/index.html) as a logs collection agent deployed on each QEMU VM with Docker containers.
```

File: docs/observability.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+* **ElasticSearch** for centralized storage and indexing of logs collected from Docker.
```
Comment (alexgarel) [inline docs/observability.md:11]:
```suggestion
* [**ElasticSearch**](https://www.elastic.co/guide/en/elasticsearch/reference/current/index.html) for centralized storage and indexing of logs collected from Docker.
```

File: docs/observability.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+* **Kibana** UI to visualize and use logs collected by ElasticSearch.
```
Comment (alexgarel) [inline docs/observability.md:13]:
```suggestion
* [**Kibana**](https://kibana.openfoodfacts.org/) UI to visualize and use logs collected by ElasticSearch. [(official doc)](https://www.elastic.co/guide/en/kibana/current/index.html)
```

File: docs/observability.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+* **Prometheus** for scraping metrics from Prometheus exporters' `/metrics` endpoint, running as sidecar containers of the applications.
```
Comment (alexgarel) [inline docs/observability.md:15]:
```suggestion
* [**Prometheus**](https://prometheus.io/docs) for scraping metrics from Prometheus exporters' `/metrics` endpoint, running as sidecar containers of the applications.
```

File: docs/observability.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+* **AlertManager** to send alerts based on Prometheus metrics, integrated with dedicated Slack channels.
+
+* **Grafana** for visualizing Prometheus metrics, InfluxDB and other metrics; and create dashboards.
```
Comment (alexgarel) [inline docs/observability.md:19]:
```suggestion
* [**Grafana**](https://grafana.openfoodfacts.org/) for visualizing Prometheus metrics, InfluxDB and other metrics; and create dashboards. [(official doc)](https://grafana.com/docs/grafana/latest)
```

File: docs/observability.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+* **Filebeat** as a logs collection agent deployed on each QEMU VM with Docker containers.
+
+* **ElasticSearch** for centralized storage and indexing of logs collected from Docker.
+
+* **Kibana** UI to visualize and use logs collected by ElasticSearch.
+
+* **Prometheus** for scraping metrics from Prometheus exporters' `/metrics` endpoint, running as sidecar containers of the applications.
+
+* **AlertManager** to send alerts based on Prometheus metrics, integrated with dedicated Slack channels.
+
```
Comment (alexgarel) [inline docs/observability.md:18]:
```suggestion

* [**InfluxDB**](https://docs.influxdata.com/influxdb/v2.0/) is the storage backend for data harvested by prometheus

```

**Review by alexgarel — APPROVED** [review]
I suggest some additions, otherwise that's really good.

File: docs/cicd.md (hunk 4 above)
Comment (ocervell) [reply docs/cicd.md:193]: making sure that our release will not break production ahead of time.
Two ways to deploy to production:
* Merge a release branch --> creates a tag vx.y.z
* Create a RC tag --> creates a tag vx.y.z-<rctag>

**Pull request merged**